src/docgen/view/documentation.ts

```diff
@@ -1,121 +1,227 @@
+  readonly readme?: boolean;
 
   /**
-   * Include a generated api reference in the documentation.
+    * Generate documentation only for a specific submodule.
+    *
+    * @default - Documentation is generated for the root module only.
+    */
+  readonly submodule?: string;
+
+}
+
+/**
+ * Options for creating a `Documentation` object.
+ */
+export interface DocumentationOptions {
+
+  /**
+   * Which language to generate docs for.
    *
-   * @default true
+   * @default 'ts'
    */
-  readonly apiReference?: boolean;
+  readonly language?: string;
 
   /**
-   * Include the user defined README.md in the documentation.
+   * Whether to ignore missing fixture files that will prevent transliterating
+   * some code snippet examples.
    *
    * @default true
    */
-  readonly readme?: boolean;
+  readonly loose?: boolean;
+}
+
+/**
+ * Options for creating a `Documentation` object using the `fromLocalPackage` function.
+ */
+export interface FromLocalPackageDocumentationOptions extends DocumentationOptions {
 
   /**
-   * Generate documentation only for a specific submodule.
+   * A local directory containing jsii assembly files that will
+   * comprise the type-system.
    *
-   * @default - Documentation is generated for all submodules.
+   * @default - the root package directory will be used.
    */
-  readonly submoduleName?: string;
+  readonly assembliesDir?: string;
 }
 
 /**
  * Render documentation pages for a jsii library.
  */
 export class Documentation {
-  private readonly assembly: reflect.Assembly;
-  private readonly submodule?: reflect.Submodule;
-
-  private readonly includeReadme: boolean;
-  private readonly includeApiReference: boolean;
-
-  private readonly transpile: Transpile;
-
-  constructor(options: DocumentationOptions) {
-    this.assembly = options.assembly;
-
-    switch (options.language) {
-      case 'python':
-        this.transpile = new PythonTranspile();
-        break;
-      case 'ts':
-        this.transpile = new TypeScriptTranspile();
-        break;
-      default:
-        throw new Error(
-          `Generating documentation for ${options.language} is not supported`,
-        );
-    }
 
-    if (options.submoduleName) {
-      this.submodule = this.findSubmodule(options.submoduleName);
+  /**
+   * Create a `Documentation` object for a remote package available in the npm registry.
+   */
+  public static async forRemotePackage(name: string, version: string, options?: DocumentationOptions): Promise<Documentation> {
+    return withTempDir(async (workdir: string) => {
+      const env = {
+        ...process.env,
+        // npm fails with EROFS if $HOME is read-only, even if it won't write there
+        HOME: os.tmpdir(),
+      };
+
+      await exec('npm install npm@7', {
+        cwd: workdir,
+        env,
+      });
+      await exec(`${workdir}/node_modules/.bin/npm install --ignore-scripts --no-bin-links --no-save ${name}@${version}`, {
+        cwd: workdir,
+        env,
+      });
```

Comment:
In this particular instance, the version with separated arguments is safer, too... And this also requires `shell` to be Windows friendly (not sure `.bin/npm` will effectively exist in Windows... it may be `npm.exe` or `npm.bat` or something... and `shell: true` is how you ensure Windows isn't stupid about those).

```suggestion
      await exec(
        `${workdir}/node_modules/.bin/npm`,
         [
           'install',
           '--ignore-scripts',
           '--no-bin-links',
           '--no-save',
           `${name}@${version}`,
         ], {
        cwd: workdir,
        env,
        shell: true,
      });
```

Reply:
Done